src/main/java/com/spotify/dns/LookupResult.java

```diff
@@ -1,19 +1,96 @@
 package com.spotify.dns;
 
-import com.google.auto.value.AutoValue;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Immutable data object with the relevant parts of an SRV record.
  */
-@AutoValue
-public abstract class LookupResult {
-  public abstract String host();
-  public abstract int port();
-  public abstract int priority();
-  public abstract int weight();
-  public abstract long ttl();
+public class LookupResult {
+
+  private final String host;
+  private final int port;
+  private final int priority;
+  private final int weight;
+  private final long ttl;
+
+  private LookupResult(final String host, final int port, final int priority, final int weight,
+                       final long ttl) {
+    this.host = checkNotNull(host, "host");
+    this.port = port;
+    this.priority = priority;
+    this.weight = weight;
+    this.ttl = ttl;
+  }
 
   public static LookupResult create(String host, int port, int priority, int weight, long ttl) {
-    return new AutoValue_LookupResult(host, port, priority, weight, ttl);
+    return new LookupResult(host, port, priority, weight, ttl);
+  }
+
+  public String host() {
+    return host;
+  }
+
+  public int port() {
+    return port;
+  }
+
+  public int priority() {
+    return priority;
+  }
+
+  public int weight() {
+    return weight;
+  }
+
+  public long ttl() {
+    return ttl;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    final LookupResult that = (LookupResult) o;
+
+    if (port != that.port) {
+      return false;
+    }
+    if (priority != that.priority) {
+      return false;
+    }
+    if (weight != that.weight) {
+      return false;
+    }
+    if (ttl != that.ttl) {
+      return false;
+    }
+    return !(host != null ? !host.equals(that.host) : that.host != null);
```

Comment:
Intellij generated code. I'm lazy but can change it if you feel strongly about it.